superflore/utils.py

```diff
@@ -235,3 +235,38 @@ def _resolve_dep_open_embedded(pkg):
         return 'poco'
     else:
         return pkg.replace('_', '-')
+
+
+def gen_delta_msg(total_changes):
+    """Return string of changes for the PR message."""
+    delta = "Changes:\n"
+    delta += "========\n"
+    for distro in total_changes:
+        if not total_changes[distro]:
+            continue
+        delta += "%s Changes:\n" % distro.title()
+        delta += "---------------\n"
+        for d in sorted(total_changes[distro]):
+            delta += '* {0}\n'.format(d)
+        delta += "\n"
+    return delta
+
+
+def gen_missing_deps_msg(missing_list):
+    """Return string of missing deps for the PR message."""
+    missing_deps = None
+    if len(missing_list) > 0:
+        missing_deps = "Missing Dependencies:\n"
+        missing_deps += "=====================\n"
+        for pkg in sorted(missing_list):
+            missing_deps += " * [ ] {0}\n".format(pkg)
+    return missing_deps or 'No missing dependencies.\n'
+
+
+def github_url_to_repo_org(url):
+    """Extract owner and repository from GitHub url."""
+    # check that the upstream_repo is a github repo
+    if 'github.com' not in url:
+        raise RuntimeError('Non-GitHub repos are not supported!')
```

Comment:
For someone reading this exception on it's own, say what is not supported. (aka extracting github org and repo name from expected github url)

Reply:
In the future this likely could be extended to support other hosting providers that also have APIs. (aka bitbucket and gitlab) It might make sense to not embed github in the function name only in the error message.

Reply:
That makes sense. Will do.